src/deploy/NVA_build/Tests.Dockerfile

```
@@ -1,21 +1,92 @@
 
 ######################
 # NOOBAA SETUP TESTS #
 ######################
 
 ENV container docker
-COPY /noobaa-core/ .
-RUN yum install -y which && \
+ENV TEST_CONTAINER true
+RUN yum install -y -q wget unzip which vim && \
+    yum group install -y -q "Development Tools" && \ 
     yum clean all
-RUN source /src/deploy/NVA_build/deploy_base.sh && \
-    install_nodejs
-RUN $(find / -name npm | head -1) install seedrandom
+RUN version="1.3.0" && \
+    wget -q http://www.tortall.net/projects/yasm/releases/yasm-${version}.tar.gz && \
+    tar xf yasm-${version}.tar.gz && \
+    pushd yasm-${version} && \
+    ./configure && \
+    make && \
+    make install && \
+    popd && \
+    rm -rf yasm-${version} yasm-${version}.tar.gz
+# In order to build this we should run docker build from the local repo
+# This layer is rebuilt when there is a new .nvmrc (when we use new version of node)
+COPY ./.nvmrc ./noobaa-core/.nvmrc
+# Install Node.js / NPM
+RUN export PATH=$PATH:/usr/local/bin && \
+    cd /usr/src && \
+    curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.31.6/install.sh | bash && \
+    export NVM_DIR="/root/.nvm" && \
+    source /root/.nvm/nvm.sh && \
+    NODE_VER=$(cat /noobaa-core/.nvmrc) && \
+    nvm install ${NODE_VER} && \
+    nvm alias default $(nvm current) && \
+    cd ~ && \
+    ln -sf $(which node) /usr/local/bin/node && \
+    ln -sf $(which npm) /usr/local/bin/npm
+WORKDIR /noobaa-core/
+# Copy the package.json and package-lock.json
+# This layer is rebuilt when there is a new package.json
+COPY ./package*.json ./
+RUN npm install
+RUN echo 'PATH=$PATH:/noobaa-core/node_modules/.bin' >> ~/.bashrc
+# Copy the binding.gyp and all the native
+# This layer is rebuilt when there a change in the native directory or in the binding.gyp
+COPY ./binding.gyp .
+COPY ./src/native ./src/native/
+RUN npm run build:native
+# Copy the frontend 
+# This layer is rebuilt when there a change in the frontend directory
+COPY ./frontend/ ./frontend/
+COPY ./src/tools/npm_install.js ./src/tools/
+RUN npm run build:fe
+
+#########################################################################################
+RUN yum install -y -q centos-release-scl && \
+    yum install -y -q rh-mongodb36 && \
+    yum install -y ntpdate && \ 
+    yum clean all
+RUN echo 'source /opt/rh/rh-mongodb36/enable' >> ~/.bashrc
+# Copy the local repo
+# This layer is rebuilt when a file changes in the project directory
+COPY . ./
+
+##########################################################################################
+
+FROM base as unitest
+
+RUN echo "const config = require('./config');" >> ./config-local.js && \
+    echo "config.SKIP_EXTERNAL_TESTS = true;" >> ./config-local.js
+
+###################################
+# Setting some test env variables #
+###################################
+# Setting cloud cradentials for the tests
+# In order to set those we need to run build with --build-arg <arg_name>="<value>"
+
+ARG aws_access_key_arg 
```

Actually, you probably don't need to have dual names - you can just expose the args that the tests already use:
```suggestion
ARG TEST_RUN_NAME
ARG AWS_ACCESS_KEY_ID
ARG AWS_SECRET_ACCESS_KEY
ARG AZURE_STORAGE_CONNECTION_STRING
```
And then you can remove them from .env because they will be in the env of any process running inside this container.

---

ARG is not getting into the env variables so, we echo it into the .env or do:
ARG aws_access_key_arg 
ENV AWS_ACCESS_KEY_ID=$aws_access_key_arg

---

Got you. resolving.